src/etcd_client.rs

```diff
@@ -46,6 +47,21 @@ impl EtcdClient {
         Ok(result_vec)
     }
 
+    /// Get key-value from etcd
+    async fn get_kv_async<T: DeserializeOwned>(&self, key: &str) -> anyhow::Result<Option<T>> {
+        let req = etcd_rs::RangeRequest::new(etcd_rs::KeyRange::key(key));
+        let mut resp =
+            self.etcd_rs_client.kv().range(req).await.map_err(|e| {
+                anyhow!("failed to get RangeResponse from etcd, the error is: {}", e)
+            })?;
```

Comment:
use context, instead of map_err, since etcd-rs has upgraded to 0.3.0

Reply:
Will change to context()?

Reply:
Done